pkg/image/importer/importer.go

```diff
@@ -459,116 +505,71 @@ func (isi *ImageStreamImporter) importRepositoryFromDocker(ctx gocontext.Context
 		if importDigest.Err != nil || importDigest.Image != nil {
 			continue
 		}
+
 		d, err := digest.ParseDigest(importDigest.Name)
 		if err != nil {
 			importDigest.Err = err
 			continue
 		}
+
+		ref := repository.Ref
+		ref.Tag = ""
+		ref.ID = string(d)
+
 		limiter.Accept()
+
 		manifest, err := s.Get(ctx, d)
 		if err != nil {
-			glog.V(5).Infof("unable to access digest %q for repository %#v: %#v", d, repository, err)
-			importDigest.Err = formatRepositoryError(repository, "", importDigest.Name, err)
+			glog.V(5).Infof("unable to get manifest by digest %q for image %s: %#v", d, ref.Exact(), err)
+			importDigest.Err = formatRepositoryError(ref, err)
 			continue
 		}
 
-		if signedManifest, isSchema1 := manifest.(*schema1.SignedManifest); isSchema1 {
-			importDigest.Image, err = schema1ToImage(signedManifest, d)
-		} else if deserializedManifest, isSchema2 := manifest.(*schema2.DeserializedManifest); isSchema2 {
-			imageConfig, getImportConfigErr := b.Get(ctx, deserializedManifest.Config.Digest)
-			if getImportConfigErr != nil {
-				glog.V(5).Infof("unable to access the image config using digest %q for repository %#v: %#v", d, repository, getImportConfigErr)
-				if isDockerError(getImportConfigErr, v2.ErrorCodeManifestUnknown) {
-					ref := repository.Ref
-					ref.ID = deserializedManifest.Config.Digest.String()
-					importDigest.Err = kapierrors.NewNotFound(imageapi.Resource("dockerimage"), ref.Exact())
-				} else {
-					importDigest.Err = formatRepositoryError(repository, "", importDigest.Name, getImportConfigErr)
-				}
-				continue
-			}
+		importDigest.Image, importDigest.Err = isi.importManifest(ctx, manifest, ref, d, s, b)
+	}
 
-			importDigest.Image, err = schema2ToImage(deserializedManifest, imageConfig, d)
-		} else {
-			// TODO: Current this error means the imported received the manifest list
-			// which we don't support yet.
-			err = fmt.Errorf("unsupported image manifest schema: %T", manifest)
-			glog.V(5).Infof("unsupported manifest type: %T", manifest)
-		}
+	doImportTag := func(importTag *importTag) bool {
+		ref := repository.Ref
+		ref.Tag = importTag.Name
+		ref.ID = ""
+
+		limiter.Accept()
 
+		manifest, err := s.Get(ctx, "", distribution.WithTag(importTag.Name))
 		if err != nil {
-			importDigest.Err = err
-			continue
+			glog.V(5).Infof("unable to get manifest by tag %q for image %s: %#v", importTag.Name, ref.Exact(), err)
+			importTag.Err = formatRepositoryError(ref, err)
+			return true
```

Comment:
"true"?  i would have guessed false but w/o code comments its hard to understand the intent either way.

Reply:
Result should be appended to `repository.Tags`.